lib/puppet/type/rds_db_option_group.rb

```diff
@@ -0,0 +1,29 @@
+Puppet::Type.newtype(:rds_db_option_group) do
+  @doc = 'Type representing an RDS DB Option group.'
+
+  newparam(:name, namevar: true) do
+    desc 'The name of the DB Option Group (also known as the db_option_group_name).'
+    validate do |value|
+      fail 'name should be a String' unless value.is_a?(String)
+    end
+  end
+
+  newproperty(:description) do
+    desc 'The description of a DB option group.'
+    validate do |value|
+      fail 'description should be a String' unless value.is_a?(String)
+    end
+  end
+
+  newproperty(:options) do
+    desc 'The options'
```

Comment:
Knowing how to set this value, or what data type `options` is expecting would help here I think.